src/analyses/base.ml

```diff
@@ -1232,6 +1232,102 @@ struct
     else
       Invariant.none
 
+  (**
+      This query returns false if the expression [exp] will definitely not result in an overflow.
+
+     Each subexpression is analyzed to see if an overflow happened.
+     For each operator in the expression, we use the query EvalInt to approximate the bounds of each
+     operand and we compute if in the worst case there could be an overflow.
+
+      For now we return true if the expression contains a shift left.
+  *)
+  let rec exp_may_signed_overflow ctx exp =
```

Comment:
This seems to duplicate the logic that already exists in `IntDomain`s. Couldn't that be reused?

Reply:
Since the `IntDomain` cannot itself evaluate expressions, some additional logic is required here. The logic in lines 1253-1261 could indeed instead delegate to the logic in the `IntDomain`. This would require changing the interface of the `IntDomTuple` to (additionally) have abstract operations that return whether the operation may have yielded an overflow. This in turn would require adding an extended `IntDomLifter` that is lifts these operations. 
So one could deduplicate some logic, but have quite some more boiler plate.